src/sentry/middleware/env.py

```diff
@@ -14,3 +16,12 @@ def process_request(self, request):
 
         # bind request to env
         env.request = request
+
+        # Periodically for an expire the local OptionsStore cache.
+        # This cleanup is purely to keep memory low and garbage collect
+        # old values. It's not required to run to keep things consistent.
+        # Internally, if an option is fetched and it's expired, it gets
+        # evicted immediately. This is purely for options that haven't
+        # been fetched since they've expired.
+        if random() < 0.25:
```

Comment:
Is `request_finished` guaranteed to execute even when raising a 500?

My thought was that a `process_request` is guaranteed to happen before anything. But if `request_finished` signal works, I can register it in there instead.